src/sheet_metal_conus.py

```diff
@@ -230,47 +232,40 @@ def effect(self):
+        path = ""
+        path += self.build_line(dictCone['ptA'], dictCone['ptB'], convFactor) # A,B
+        path += " " + self.build_arc(zeroCenter, 0.0, angle, dictCone['longRadius']*convFactor)
+        path += " " + self.build_line(dictCone['ptC'], dictCone['ptD'], convFactor) # C,D
+        path += self.build_arc(zeroCenter, angle, 0.0, dictCone['shortRadius']*convFactor, False)
+        line_attribs['d'] = path
+        ell = inkex.etree.SubElement(grp, inkex.addNS('path','svg'), line_attribs )
         
         # Draw Dimensions Markup
         if self.options.verbose == True:
             grp_attribs = {inkex.addNS('label','inkscape'):'markup'}
             markup_group = inkex.etree.SubElement(grp, 'g', grp_attribs)
             self.beVerbose(dictCone, convFactor, markup_group)
                 
- 
-    def drawEllipse(self, radiusX, radiusY, (centerX, centerY), startAngle, endAngle, style, label, unitFactor, parent):
-        """ Draws an ellipse (or arc if radiusX=radiusY) using
-            - center point, start angle, end angle, and a given style.
-            unitfactor scales all dimensions.
-        """
-        attribs = {'style' : simplestyle.formatStyle(style),
-        inkex.addNS('label','inkscape') : label,
-        inkex.addNS('cx','sodipodi')     :str(centerX * unitFactor),
-        inkex.addNS('cy','sodipodi')     :str(centerY * unitFactor),
-        inkex.addNS('rx','sodipodi')     :str(radiusX * unitFactor),
-        inkex.addNS('ry','sodipodi')     :str(radiusY * unitFactor),
-        inkex.addNS('start','sodipodi')  :str(startAngle),
-        inkex.addNS('end','sodipodi')    :str(endAngle),
-        inkex.addNS('open','sodipodi')   :'true',    #all ellipse sectors we will draw are open
-        inkex.addNS('type','sodipodi')   :'arc' }
-        ell = inkex.etree.SubElement(parent, inkex.addNS('path','svg'), attribs )
-
-    def drawLine(self, (x1, y1), (x2, y2), style, label, unitFactor, parent):
-        """ Draws a line from given point 1 to given point 2 with a given style.
-        """
-        line_attribs = {'style' : simplestyle.formatStyle(style), inkex.addNS('label','inkscape') : label,
-                        'd' : 'M %s,%s L %s,%s' % (x1*unitFactor, y1*unitFactor, x2*unitFactor, y2*unitFactor)}
-        line = inkex.etree.SubElement(parent, inkex.addNS('path','svg'), line_attribs )
+    def build_arc(self, (x,y), start_angle, end_angle, radius, reverse=True):
+        " Make an arc "
+        # Not using internal arc rep - instead construct path A in svg style directly
+        # so we can append lines to make single path
+        start = polar_to_cartesian(x, y, radius, end_angle)
+        end = polar_to_cartesian(x, y, radius, start_angle)
+        arc_flag = 0 if reverse else 1
+        sweep = 0 if (end_angle-start_angle) <=180 else 1
+        path = 'M %s,%s' % (start[0], start[1])
+        path += " A %s,%s 0 %d %d %s %s" % (radius, radius, sweep, arc_flag, end[0], end[1])
+        return path
```

Comment:
Build_arc() is a replacement for drawEllpse() and returns a  path instead of building an entity.